test/m1_facilitator/handlers/DeclaredDepositIntentsHandler/handle.test.ts

```diff
@@ -0,0 +1,198 @@
+    const gateway = new Gateway(
+      record.contractAddress,
+      '0x0000000000000000000000000000000000000010',
+      GatewayType.CONSENSUS,
+      '0x0000000000000000000000000000000000000001',
+      '0x0000000000000000000000000000000000000030',
+      new BigNumber(200),
+    );
+    await gatewayRepository.save(
+      gateway,
+    );
+
+    declaredDepositIntentsHandler = new DeclaredDepositIntentsHandler(
+      repositories.depositIntentRepository,
+      repositories.gatewayRepository,
+      repositories.messageRepository,
+    );
+  });
+
+  it('should change source status of message from undeclared to declared if message'
+    + ' is not present', async (): Promise<void> => {
+    await declaredDepositIntentsHandler.handle([record]);
+
+    const message = await messageRepository.get(record.messageHash);
+
+    assert.strictEqual(
+      message && message.type,
+      MessageType.Deposit,
+      'Message type should be deposit',
+    );
+
+    assert.strictEqual(
+      message && message.sourceStatus,
+      MessageStatus.Declared,
+      'Source status must be declared',
+    );
+
+    assert.strictEqual(
+      message && message.targetStatus,
+      MessageStatus.Undeclared,
+      'Target status must be undeclared',
+    );
+
+    const depositIntent = await depositIntentRepository.get(record.messageHash);
+
+    assert.isOk(
+      depositIntent,
+      `Record must be present for ${record.messageHash} in deposit intent model`,
+    );
+  });
+
+  it('should create a deposit intent if message is not present is'
+  + ' not present', async (): Promise<void> => {
+    await declaredDepositIntentsHandler.handle([record]);
+
+    const depositIntent = await depositIntentRepository.get(record.messageHash);
+
+    assert.isOk(
+      depositIntent,
+      `Record must be present for ${record.messageHash} in deposit intent model`,
+    );
+  });
+
+  it('should change status of message from undeclared to declared for'
+    + ' existing message', async (): Promise<void> => {
+    await messageRepository.save(
+      new Message(
+        record.messageHash,
+        MessageType.Deposit,
+        MessageStatus.Undeclared,
+        MessageStatus.Undeclared,
+        '0x0000000000000000000000000000000000000050',
+        new BigNumber(0),
+        new BigNumber(0),
+        new BigNumber(0),
+        record.intentHash,
+      ),
+    );
+
+    await declaredDepositIntentsHandler.handle([record]);
+
+    const message = await messageRepository.get(record.messageHash);
+
+    assert.strictEqual(
+      message && message.sourceStatus,
+      MessageStatus.Declared,
+      'Source status must be declared',
+    );
+
+    assert.strictEqual(
+      message && message.targetStatus,
+      MessageStatus.Undeclared,
+      'Target status must be undeclared',
+    );
+  });
+
+  it('should handle multiple records', async (): Promise<void> => {
+    const record2 = {
```

Comment:
we can call it record1, record2. Similary message1, message2.
This will be much cleaner in overall tests.

Reply:
Done